validator/validation/validation.py

```diff
@@ -58,6 +59,21 @@ def _get_actual_time_range(val_run, dataset_version_id):
 
     return [actual_start, actual_end]
 
+def _get_reference_reader(val_run):
+    ref_reader = create_reader(val_run.reference_configuration.dataset, val_run.reference_configuration.version)
+
+    # we do the dance with the filtering below because filter may actually change the original reader, see ismn network selection
+    ref_reader = setup_filtering(ref_reader, list(val_run.reference_configuration.filters.all()),\
+                                 list(val_run.reference_configuration.parametrisedfilter_set.all()),\
+                                 val_run.reference_configuration.dataset, val_run.reference_configuration.variable)
+    while(hasattr(ref_reader, 'cls')):
+        ref_reader = ref_reader.cls
+
+    if not isinstance(ref_reader, ISMN_Interface):
+        global METADATA_TEMPLATE
+        METADATA_TEMPLATE = None
```

Comment:
Instead of manipulating a global variable I would suggest making `METADATA_TEMPLATE` a dictionary of metadata templates and then choose the value to use in the calls to the metrics calculators.